apps/bluetooth/js/views/device_picker_panel.js

```diff
@@ -0,0 +1,410 @@
+    _onPairedDeviceItemClick:
+    function btdpp__onPairedDeviceItemClick(deviceItem) {
+      Debug('_onPairedDeviceItemClick(): deviceItem.address = ' +
+            deviceItem.address);
+      Debug('_onPairedDeviceItemClick(): deviceItem.paired = ' +
+            deviceItem.paired);
+      // Emit 'devicePicked' event to outer modules with the device item.
+      var event = {
+        type: 'devicePicked',
+        detail: {
+          address: deviceItem.address
+        }
+      };
+      this._emitEvent(event);
+    },
+
+    /**
+     * Pair with the clicked device. While the device is paired successfully,
+     * emit event to outer modules with the device item.
+     *
+     * @access private
+     * @memberOf DevicePickerPanel
+     */
+    _onFoundDeviceItemClick:
+    function btdpp__onFoundDeviceItemClick(deviceItem) {
+      Debug('_onFoundDeviceItemClick(): deviceItem.address = ' +
+            deviceItem.address);
+      // Update device pairing status first.
+      deviceItem.paired = 'pairing';
+      // Pair with the remote device.
+      BtContext.pair(deviceItem.address).then(() => {
+        Debug('_onFoundDeviceItemClick(): pair successfully');
+        // Emit 'devicePicked' event to outer modules with the 
+        // paired device item.
+        var event = {
+          type: 'devicePicked',
+          detail: {
+            address: deviceItem.address
+          }
+        };
+        this._emitEvent(event);
+      }, (reason) => {
+        Debug('_onFoundDeviceItemClick(): pair failed, ' +
+              'reason = ' + reason);
+        // Reset the paired status back to false, 
+        // since the 'pairing' status is given in Gaia side.
+        deviceItem.paired = false;
+        // Show alert message while pair device failed.
+        this._alertPairFailedErrorMessage(reason);
+      });
+    },
+
+    /**
+     * Show an alert dialog since the device is failed in pairing process.
+     *
+     * @access private
+     * @memberOf DevicePickerPanel
+     */
+    _alertPairFailedErrorMessage:
+    function btdpp__alertPairFailedErrorMessage(errorReason) {
+      var _ = navigator.mozL10n.get;
+      var msg = _('error-pair-title'); // This is a default message.
+      if (errorReason === 'Repeated Attempts') {
+        msg = msg + '\n' + _('error-pair-toofast');
+      } else if (errorReason === 'Authentication Failed') {
+        msg = msg + '\n' + _('error-pair-pincode');
+      }
+      window.alert(msg);
+    },
+
+    /**
+     * A function to show/hide paired devices header.
+     *
+     * @access private
+     * @memberOf DevicePickerPanel
+     */
+    _updatePairedDevicesHeader:
+    function btdpp__updatePairedDevicesHeader(hasPairedDevice) {
+      Debug('_updatePairedDevicesHeader(): ' +
+            'callback from observe "hasPairedDevice" = ' + hasPairedDevice);
+      this._elements.paired.pairedDevicesHeader.hidden =
+        (hasPairedDevice && (BtContext.state === 'enabled')) ? false : true;
```

Comment:
nit: simply `this._elements.paired.pairedDevicesHeader.hidden = !(hasPairedDevice && (BtContext.state === 'enabled'));`